File: backend/src/Playbook/McpExecutorInterface.php
<?php
declare(strict_types=1);
namespace Quantis\AIPortfolioAssistant\Playbook;

interface McpExecutorInterface
{
    /** @return array{ok:bool,result?:mixed,error?:string} */
    public function call(string $server, string $tool, array $args): array;

    /** @return array<string,array{description:string,input_schema:array}> keyed "server.tool" */
    public function availableTools(): array;
}

File: backend_python/app/playbook/mcp_executor_interface.py
"""Port of backend/src/Playbook/McpExecutorInterface.php (12 lines)."""
from __future__ import annotations

from abc import ABC, abstractmethod


class McpExecutorInterface(ABC):
    @abstractmethod
    def call(self, server: str, tool: str, args: dict) -> dict:
        """@return {ok: bool, result?: mixed, error?: str}"""
        raise NotImplementedError

    @abstractmethod
    def availableTools(self) -> dict:
        """@return {"server.tool": {description: str, input_schema: dict}}"""
        raise NotImplementedError
